scripts/release/generate_changelog.sh

```diff
@@ -13,34 +13,33 @@ set -euo pipefail
 function step { >&2 echo -e "\033[1m\033[36m* $@\033[0m"; }
 function finish { >&2 echo -en "\033[0m"; }
 
+# Branch that is passed as input
+BRANCH=${1}
+
 # Guarantees installation of release tools
 npm install -g @mapbox/github-release-tools
 
 # Base branch to compare to and where we want to commit our changes
-# TODO: Future work to adjust branch to release strategy
-git checkout main
+git checkout ${BRANCH}
 
-# Remove any existing tags if there are any, and then fetch all remote tags
+# Remove any existing local tags if there are any, and then fetch all remote tags
 git tag -d $(git tag -l)
 git fetch --tags
 
 # Get all tags matching '10.0.0*' in order of latest activity
 # TODO: Modularize this so that it is not hardcoded to version 10.0.0
-ORDERED_TAGS=`git tag -l --sort=-committerdate -l "10.0.0*"`
+ORDERED_TAGS=`git tag -l --sort=-committerdate -l "v10.0.*"`
```

Comment:
This isn't a valid assumption right? What happens if there's a patch release for `beta.12` for example?

Reply:
correct, I do have a TODO there to modularize it.

To your point though, this script is designed to compare w.e branch you provide it, to the last tag. So if we did a patch to beta 12, we wouldn't want to compare that to like beta 14.

Reply:
Also beta 12(public) onwards all have the v prefix. So this change would only break for private beta patches